### PR TITLE
Do not make unmarshaling errors block the metrics migration.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by rogpeppe** [comment]
LGTM
Old charms won't have metrics anyway.

File: internal/charmstore/migrations.go [hunk 1]
```diff
@@ -499,7 +499,10 @@ func getEntityMetrics(blobStore *blobstore.Store, entity mongodoc.Entity) (*char
 	defer r.Close()
 	ch, err := charm.ReadCharmArchiveFromReader(ReaderAtSeeker(r), entity.Size)
 	if err != nil {
-		return nil, errgo.Notef(err, "cannot read charm archive bytes")
+		// Some of the entities in the store are no longer readable due to
+		// unmarshaling differences between yaml.v1 and yaml.v2, in which the
+		// former allowed type mismatches when deserielizing empty values.
+		return nil, nil
```
Comment (mhilton) [inline internal/charmstore/migrations.go:506]: If you're going to do that can you log the one's we're skipping?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1828/
Test PASSed.

**Comment by mhilton** [comment]
:+1: but with a request

File: internal/charmstore/migrations.go [hunk 2]
```diff
@@ -499,7 +499,11 @@ func getEntityMetrics(blobStore *blobstore.Store, entity mongodoc.Entity) (*char
 	defer r.Close()
 	ch, err := charm.ReadCharmArchiveFromReader(ReaderAtSeeker(r), entity.Size)
 	if err != nil {
-		return nil, errgo.Notef(err, "cannot read charm archive bytes")
+		// Some of the entities in the store are no longer readable due to
+		// unmarshaling differences between yaml.v1 and yaml.v2, in which the
+		// former allowed type mismatches when deserielizing empty values.
+		logger.Errorf("skipping %q: error reading the charm: %q", entity.URL, err)
```
Comment (mhilton) [inline internal/charmstore/migrations.go]: The second %q could be a %s (I think) but it's probably not going to add much

Comment (frankban) [inline internal/charmstore/migrations.go]: Done.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1829/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1830/
Test PASSed.

**Pull request merged**